vsphere_plugin_common/__init__.py

```diff
@@ -433,14 +433,25 @@ def _get_entity(self, entity_name, props, vimtype, use_cache=True,
                     )
                 )
             except KeyError as err:
-                if not skip_broken_objects:
-                    raise NonRecoverableError(
-                        'Could not retrieve all details for {type} object. '
-                        '{err} was missing.'.format(
-                            type=entity_name,
-                            err=str(err)
-                        )
+                message = (
+                    'Could not retrieve all details for {type} object. '
+                    '{err} was missing.'.format(
+                        type=entity_name,
+                        err=str(err)
+                    )
+                )
+                if hasattr(result, 'name'):
+                    message += (
+                        ' Object name was {name}.'.format(name=result.name)
+                    )
+                if hasattr(result, '_moId'):
+                    message += (
+                        ' Object ID was {id}.'.format(id=result._moId)
                     )
+                if skip_broken_objects:
+                    ctx.logger.warn(message)
```

Comment:
Yeah, I have a faint suspicion that we're going to need this for more than just hosts... not 100% certain of that yet, but I think this is a step towards us being able to just set skip_broken_objects to true by default for everything.